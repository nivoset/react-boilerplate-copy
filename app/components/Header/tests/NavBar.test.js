import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import NavBar from '../NavBar';

const renderComponent = (props = {}) => {
  const { container } = render(<NavBar {...props} />);
  return container;
};

describe('<NavBar />', () => {
  it('should match the snapshot', () => {
    const container = renderComponent();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should have a class attribute', () => {
    const container = renderComponent();
    expect(container.firstChild).toHaveAttribute('class');
  });

  it('should not adopt an invalid attribute', () => {
    const container = renderComponent({ attribute: 'test' });
    expect(container.firstChild).not.toHaveAttribute('attribute');
  });
});
