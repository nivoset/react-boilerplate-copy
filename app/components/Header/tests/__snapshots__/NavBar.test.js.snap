// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> should match the snapshot 1`] = `
.c0 {
  text-align: center;
}

<div
  class="c0"
/>
`;
